src/etl_image.cpp

```diff
@@ -92,6 +92,10 @@ shared_ptr<image::decoded> image::extractor::extract(const void* inbuf, size_t i
     cv::Mat input_img(1, insize, _pixel_type, (char*)inbuf);
     cv::imdecode(input_img, _color_mode, &output_img);
 
+    if (output_img.data == nullptr) {
+        throw runtime_error("Input image contains invalid data or the buffer is too short.");
+    }
```

Comment:
what do you mean by comment - buffer is too short ? which buffer ?

Reply:
I know its not quite intuitive, it relates to `const void* inbuf` so usually it is what was read from the image file. I'm open to suggestions for an easier to understand message.

The original information is this:
>The function reads an image from the specified buffer in the memory. If the buffer is too short or contains invalid data, the empty matrix/image is returned.

https://docs.opencv.org/3.0-beta/modules/imgcodecs/doc/reading_and_writing_images.html?highlight=imread

Reply:
1. instead of use pointer comparison output_img.data == nullptr we can use appropriate function output_img.empty()
2. This message is for user, so I would suggest to leave only first part of message "Input image contains invalid data", because in other case user start thinking about some buffer, and will be trying to find why some unknown buffer that is too short, but in this case we have problem only with corrupted jpeg

Reply:
I changed the message a bit. Please check it again.